db/db_iter.cc

```diff
@@ -930,6 +965,16 @@ void DBIter::FindPrevUserKey() {
   }
 }
 
+bool DBIter::TooManyInternalKeysSkipped() {
+  if ((max_skippable_internal_keys_ > 0) &&
+      (num_internal_keys_skipped_ > max_skippable_internal_keys_)) {
+    valid_ = false;
+    status_ = Status::Incomplete("Too many internal keys skipped.");
+    return true;
+  }
+  return false;
+}
+
```

Comment:
I definitely did consider this, and I would have loved to done the exact same thing which you suggested, but it will not work in the reverse iteration flow as we will end up double counting in a few cases. 
The Reason being: 
TooManyInternalKeysSkipped is called from both PrevInternal and  FindValueForCurrentKey. FindValueForCurrentKey is in turn called from PrevInternal. If the counter is incremented in FindValueForCurrentKey, it should not be incremented again in PrevInternal. One of my unit tests ran into this problem of double counting, and I had to fix it.